enterprise/app/controllers/api/v1/accounts/captain/documents_controller.rb

```diff
@@ -3,6 +3,7 @@ class Api::V1::Accounts::Captain::DocumentsController < Api::V1::Accounts::BaseC
   before_action -> { check_authorization(Captain::Assistant) }
 
   before_action :set_current_page, only: [:index]
+  before_action :ensure_within_plan_limit, only: [:create]
```

Comment:
should we do this in the model itself to ensure that even the crawl job would stay under the limit

Reply:
Should we throw an error there?